### DISCUSSION TITLE
Feature request: External API Key Access

### DISCUSSION THREAD

**Discussion by aliasfoxkde** [oobabooga/text-generation-webui#1017] [Ideas]
I would personally like to use BetterChatGPT UI with the oobabooga/text-generation-webui but it requires an API Key to setup. Not sure which direction would be best but I think it would be useful to have the thing running the model expose an API Key and endpoints. I think an API Key to access the AI model would be useful for extending the capabilities to other applications, possibly extending into decentralized or load-balancing multiple instances, and generally just make scaling easier. Though I have a specific use case in mind, I think adding this feature would only extend it's usefulness.

Also, I looked but saw nothing on this, so if it's a duplicate or I overlooked this feature in documentation, someone please correct and direct me.

Thanks you!

**Comment by personthe** [comment]
hard agree

**Comment by swizzcheeze** [comment]
AGREE

**Comment by simcop2387** [comment]
Take a look in the wiki, this is how you set it up: https://github.com/oobabooga/text-generation-webui/wiki/12-%E2%80%90-OpenAI-API#third-party-application-setup

**Reply by aliasfoxkde** [reply]
Amazing. Thanks!
